### ISSUE TITLE
Show only billable clients and projects

### ISSUE THREAD

**Issue by Shivam-Samant** [ColoredCow/portal#3441]
**Summary of Feature Request:**
In the invoice creating and listing screen billable and non-billable clients/projects are shown, but in that non-billable clients/projects are not used so needed to remove those unnecessary details as they are not utilized in those contexts.

**Solution**
- When creating a project, add a new project type called `non-billable` and use that to hide all the non-billable projects in the invoice creation and listing screen.
- Create the `is_billable` column to determine the billable and non-billable clients, and using that hide all the non-billable clients from the invoice creation and listing screen.

**Approach**
- Add Non Billable project type in the create/edit project screen.
- Create migration to add the `is_billable` column in the `clients` table and make the default value `true`.
- Add the checkbox for the `is_billable` column in the create/edit client screen.
- Use the non-billable project type to hide the non-billable projects in the invoice create/edit screen.
- Use the `is_billable` column of the `clients` table to only show the billable clients in the invoice create/edit screen.

**Comment by P4NK4J** [comment]
@Shivam-Samant  The requirements are fine, just one question, there might be clients for whom we have nonbillable projects as well. We need to hide all the nonbillable projects from the invoice screen if a client is billable. 
 If a client is not billable, then it should not be displayed at all. Let me know if this would be addressed here as well.

**Comment by Shivam-Samant** [comment]
> @Shivam-Samant The requirements are fine, just one question, there might be clients for whom we have nonbillable projects as well. We need to hide all the nonbillable projects from the invoice screen if a client is billable. If a client is not billable, then it should not be displayed at all. Let me know if this would be addressed here as well.

Yes, so if the client is billable or partially billable then we mark that client as billable but in case of the partially billable client, the system will hide its non billable projects.
